apiv2/views.py

```diff
@@ -135,14 +135,16 @@ def get(self, request,  *args, **kwargs):
                                                                        page=page['next_page_number'])
 
         # Get analysis data and serialize sound results
-        ids = [id for id in page['object_list']]
-        get_analysis_data_for_queryset_or_sound_ids(self, sound_ids=ids)
-        sounds_dict = Sound.objects.dict_ids(sound_ids=ids)
+        object_list = [ob for ob in page['object_list']]
+        id_score_map = dict(object_list)
+        sound_ids = [ob[0] for ob in object_list]
+        get_analysis_data_for_queryset_or_sound_ids(self, sound_ids=sound_ids)
+        sounds_dict = Sound.objects.dict_ids(sound_ids=sound_ids)
 
         sounds = []
-        for i, sid in enumerate(ids):
+        for i, sid in enumerate(sound_ids):
             try:
-                sound = SoundListSerializer(sounds_dict[sid], context=self.get_serializer_context()).data
+                sound = SoundListSerializer(sounds_dict[sid], context=self.get_serializer_context(), score_map=id_score_map).data
```

Comment:
This is a nice way of passing extra info to the serializer. Same technique should be used for passing analysis data for the `analysis` field. The way I did it until now was to save this data in the view (see ugly code in `get_analysis_data_for_queryset_or_sound_ids`) and then pass the context to the serializer so it can retrieve a reference to the view. However latetly I've been a bit suspicious about that code, thinking that it might lead to mysterious issues in the api when getting analysis data (one day I told you we should talk about these issues). So I think I should follow that same strategy you propose. After merging this PR, I can work on my changes.